File: src/main/java/com/hopper/model/availability/agoda/PolicyDate.java
package com.hopper.model.availability.agoda;

import javax.annotation.Nullable;
import javax.xml.bind.annotation.*;

@XmlRootElement(name="PolicyDate")
@XmlAccessorType(XmlAccessType.FIELD)
public class PolicyDate {


    @Nullable
    @XmlAttribute(name="before", required = false)
    private String before;

    @Nullable
    @XmlAttribute(name="after",required = false)
    private String after;

    @XmlElement(name = "Rate")
    private Rate rate;

    public String getAfter() {
        return after;
    }

    public void setAfter( String after) {
        this.after = after;
    }

    public String getBefore() {
        return before;
    }

    public void setBefore( String before) {
        this.before = before;
    }

    public Rate getRate() {
        return rate;
    }

    public void setRate(Rate rate) {
        this.rate = rate;
    }

    @Override
    public String toString() {
        return "PolicyDate{" +
                "before='" + before + '\'' +
                ", after='" + after + '\'' +
                ", rate=" + rate +
                '}';
    }
}

File: src/main/scala/com/hopper/model/agoda/availability/response/PolicyDate.scala
package com.hopper.model.agoda.availability.response

import javax.annotation.Nullable
import javax.xml.bind.annotation.{XmlAccessType, XmlAccessorType, XmlAttribute, XmlElement, XmlRootElement}

@XmlRootElement(name = "PolicyDate")
@XmlAccessorType(XmlAccessType.NONE)
class PolicyDate
{
    @Nullable
    @XmlAttribute(name = "before", required = false)
    var before: String = _

    @Nullable
    @XmlAttribute(name = "after", required = false)
    var after: String = _

    @XmlElement(name = "Rate")
    var rate: Rate = _
}